src/Functions/array/arrayAUC.cpp

```diff
@@ -131,56 +133,78 @@ class FunctionArrayAUC : public IFunction
 
         area += (curr_fp - prev_fp) * (curr_tp + prev_tp) / 2.0;
 
-        /// Then normalize it dividing by the area to the area of rectangle.
+        /// Then normalize it, if scale is true, dividing by the area to the area of rectangle.
 
-        if (curr_tp == 0 || curr_tp == size)
+        if (scale && (curr_tp == 0 || curr_tp == size))
```

Comment:
Could we just write

```cpp
if (scale)
{
    if (curr_tp == 0 || curr_tp == size)
        return std::numeric_limits<Float64>::quiet_NaN();
    return area / curr_tp / (size - curr_tp);
}
else
{
    return area;
}

Reply:
is an early return cleaner here?

```c++
        if (scale)
        {
            if (curr_tp == 0 || curr_tp == size)
                return std::numeric_limits<Float64>::quiet_NaN();
            return area / curr_tp / (size - curr_tp);
        }
        return area;
```

Reply:
Early result or not doesn't matter too much but AND-ing `scale` in two if conditions looks a bit ugly.